### PR TITLE
Add support for try=chooser syntax

### PR DESCRIPTION
**Pull request by Manishearth**
Adds support for try choosers in the configuration.

SQLite migration notes: Either start from a fresh db, or run `ALTER TABLE pull ADD try_choose text`

---
This change is [Reviewable](https://reviewable.io/reviews/servo/homu/167)

### CODE REVIEW

File: homu/main.py [hunk 1]
```diff
@@ -506,6 +507,11 @@ def parse_commands(cfg, body, username, repo_cfg, state, my_username, db,
                 continue
             action._try(state, word)
 
+        elif wor.startswith("try=") and realtime:
```
Comment (jdm) [inline homu/main.py]: `word`, not wor, and indentation is off.

File: homu/main.py [hunk 2]
```diff
@@ -1447,9 +1459,9 @@ def main():
 
         db_query(
             db,
-            'SELECT num, head_sha, status, title, body, head_ref, base_ref, assignee, approved_by, priority, try_, rollup, delegate, merge_sha FROM pull WHERE repo = ?',   # noqa
+            'SELECT num, head_sha, status, title, body, head_ref, base_ref, assignee, approved_by, priority, try_, try_choose rollup, delegate, merge_sha FROM pull WHERE repo = ?',   # noqa
```
Comment (jdm) [inline homu/main.py]: Missing a , after try_choose.

**Comment by Manishearth** [comment]
So homu doesn't actually kick off these builds -- buildbot does. Homu just waits for completion pings for these builds.

So we need to add a `SingleBranchScheduler` for branches named try-foo and try-bar. Does this sound ok?

**Pull request merged**